docs/proposals/formatted-output.md

```diff
@@ -0,0 +1,358 @@
+- `digest`: digest of the image manifest
+- `size`: manifest file size in bytes
+- `artifactType`: the type of an artifact when the manifest is used for an artifact
+- `content`: content output includes descriptor metadata and layers in filesystem
+    - `config`: a content descriptor describes the disposition of the targeted content
+    - `layers`:  array of objects, each object in the array MUST be a descriptor
+
+See example use cases of formatted output for `oras manifest fetch`:
+
+- Use `--output <file>` and `--format` at the same time, a manifest file should be produced in the filesystem and the `mediaType` value should be outputted on the console:
+
+```console
+$ oras manifest fetch $REGISTRY/$REPO:$TAG --output sample-manifest --format {{.config.mediaType}}
+application/vnd.oci.empty.v1+json
+```
+
+View the contents of the generated manifest within specified `sample-manifest` file. The contents should be raw JSON data:
+
+```console
+$ cat sample-manifest
+{"schemaVersion":2,"mediaType":"application/vnd.oci.image.manifest.v1+json","artifactType":"application/vnd.unknown.artifact.v1","config":{"mediaType":"application/vnd.oci.empty.v1+json","digest":"sha256:44136fa355b3678a1146ad16f7e8649e94fb4fc21fe77e8310c060f61caaff8a","size":2,"data":"e30="},"layers":[{"mediaType":"application/vnd.oci.image.layer.v1.tar","digest":"sha256:6cb759c4296e67e35b0367f3c0f51dfdb776a0c99a45f39d0476e43d82696d65","size":14477,"annotations":{"org.opencontainers.image.title":"sbom.spdx"}},{"mediaType":"application/vnd.oci.image.layer.v1.tar","digest":"sha256:54c0e84503c8790e03afe34bfc05a5ce45c933430cfd9c5f8a99d2c89f1f1b69","size":6639,"annotations":{"org.opencontainers.image.title":"scan-test-verify-image.json"}}],"annotations":{"org.opencontainers.image.created":"2023-12-15T09:41:54Z"}}
+```
+
+- Use `--format json` to print the output in prettified JSON:
+
+```bash
+oras manifest fetch $REGISTRY/$REPO:$TAG --format json
+```
+
+```json
+{
+  "schemaVersion": 2,
+  "mediaType": "application/vnd.oci.image.manifest.v1+json",
+  "artifactType": "application/vnd.unknown.artifact.v1",
+  "config": {
+    "mediaType": "application/vnd.oci.empty.v1+json",
+    "digest": "sha256:44136fa355b3678a1146ad16f7e8649e94fb4fc21fe77e8310c060f61caaff8a",
+    "size": 2,
+    "data": "e30="
+  },
+  "layers": [
+    {
+      "mediaType": "application/vnd.oci.image.layer.v1.tar",
+      "digest": "sha256:6cb759c4296e67e35b0367f3c0f51dfdb776a0c99a45f39d0476e43d82696d65",
+      "size": 14477,
+      "annotations": {
+        "org.opencontainers.image.title": "sbom.spdx"
+      }
+    },
+    {
+      "mediaType": "application/vnd.oci.image.layer.v1.tar",
+      "digest": "sha256:54c0e84503c8790e03afe34bfc05a5ce45c933430cfd9c5f8a99d2c89f1f1b69",
+      "size": 6639,
+      "annotations": {
+        "org.opencontainers.image.title": "scan-test-verify-image.json"
+      }
+    }
+  ],
+  "annotations": {
+    "org.opencontainers.image.created": "2023-12-15T09:41:54Z"
+  }
+}
+```
```

Comment:
This manifest json is wrong, the displayed data is only a partial data of `{{.contents}}`.

Reply:
Updated and added `content`